cpp/tests/test_ide_secir.cpp

```diff
@@ -242,15 +242,181 @@ TEST(IdeSecir, checkSimulationFunctions)
     }
 }
 
+// Check if the model uses uses the correct method for initialization.
+// This is done by comparing the compartment sizes at t0 with a previous run with the same initialization method.
```

Comment:
Idea was to target all the method for initialization. There is no way to get which method was used to initialize, so I simply compared the results to a previous run. The idea is that the same method was used when we have same results. If the results don't match, something has gone wrong. I created the results with a simulation in which I checked (via debug mode) which method was used. Are you okay with this idea or maybe a better suggestion?

Reply:
Thanks for the explanation. The issue I see here is that the test depends on the results of the respective initialization method although what is tested is not if these results are correct but that the correct if/else statement was selected. This way the test needs to be maintained and changed every time an initialization method is changed (although I don't see how this would change in the near future).
One idea to avoid this would be to set m_total_confirmed_cases =-1,-2 etc. (or just some value that is not valid for the initialization method using m_total_confirmed_cases) depending on the initialization option and to check the value of m_total_confirmed_cases in the test.

Reply:
I rewrote the test and added a function to the class to get the method for initialization.